### ISSUE TITLE
Brutile HttpTileSource is not compatible to WebAssembly

### ISSUE THREAD

**Issue by inforithmics** [BruTile/BruTile#168]
Current Code in HttpTileSource which does not work in WebAssembly (Throws an exception)

```
private byte[] FetchTile(Uri arg)
{
    return _httpClient.GetByteArrayAsync(arg).ConfigureAwait(false).GetAwaiter().GetResult();
}
```

Code that would work in HttpTileSource in WebAsdembly.

```
private async Task<byte[]> FetchTileAsync(Uri arg)
{
    return await _httpClient.GetByteArrayAsync(arg).ConfigureAwait(false);
}
```

This would cause to change the Public Api Of Api (Async Calls where there where before Sync Calls).
Two Possible Solutions.

1. Create an AsyncHttpTileSource
2. Change the Api of HttpTileSource and increase the Brutile Main Version to 5.0.0

Sample where this does not work:
https://github.com/Mapsui/Mapsui/pull/1525 
Mapsui.Samples.Uno.Wasm.csproj

StackOverflow Question:
https://stackoverflow.com/questions/63211539/blazor-startup-error-system-threading-synchronizationlockexception-cannot-wait

**Comment by pauldendulk** [comment]
Thanks. I looked at the code and I think the best option is nr 2, replace it and release 5.0.0. 

"it’s usually a bad idea to block on async code by calling Task.Wait or Task.Result" source: https://docs.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming#async-all-the-way

Also supporting wasm is important to me. I would like to have our samples to run in wasm.

**Comment by pauldendulk** [comment]
Fixed with #169
